src/main/java/com/splunk/logging/HttpEventCollectorLogbackAppender.java

```diff
@@ -51,31 +51,32 @@
     private long _batchSize = 0;
     private String _sendMode;
     private long _retriesOnError = 0;
-
+    private Map<String, String> _metadata = new HashMap<>();
+    
+    
     @Override
     public void start() {
         if (started)
             return;
 
         // init events sender
-        Map<String, String> metadata = new HashMap<>();
```

Comment:
By putting this into the object scope, this means that the values from one event can affect the next event- for example if one event sets the index and the next doesn't, the second event will "inherit" the message for the first. Could you add this line back in either as `new HashMap<>(this._metadata)` or `this._metadata.clone()`?